desc/magnetic_fields.py

```diff
@@ -795,3 +831,464 @@ def odefun(rpz, s):
+        assert np.isscalar(G), "G must be a scalar"
+        self._I = I
+        self._G = G
+
+        def _fourier_potential(theta, zeta, I, G, Phi_mn):
+            # theta, zeta are eval points on the winding surface
+            nodes = jnp.vstack(
+                (
+                    jnp.zeros_like(theta.flatten(order="F")),
+                    theta.flatten(order="F"),
+                    zeta.flatten(order="F"),
+                )
+            ).T
+            trans_temp = Transform(
+                Grid(nodes, sort=False, jitable=True),
+                basis,
+            )
+            return (
+                trans_temp.transform(Phi_mn, jitable=True)
+                + G * zeta.flatten(order="F") / 2 / jnp.pi
+                + I * theta.flatten(order="F") / 2 / jnp.pi
+            )
+
+        def _fourier_potential_d_theta(theta, zeta, I, G, Phi_mn):
+            # theta, zeta are eval points on the winding surface
+            nodes = jnp.vstack(
+                (
+                    jnp.zeros_like(theta.flatten(order="F")),
+                    theta.flatten(order="F"),
+                    zeta.flatten(order="F"),
+                )
+            ).T
+            trans_temp = Transform(
+                Grid(nodes, sort=False, jitable=True),
+                basis,
+                derivs=jnp.array([[0, 1, 0]]),
+            )
+            return trans_temp.transform(Phi_mn, dt=1) + I / 2 / jnp.pi
+
+        def _fourier_potential_d_zeta(theta, zeta, I, G, Phi_mn):
+            # theta, zeta are eval points on the winding surface
+            nodes = jnp.vstack(
+                (
+                    jnp.zeros_like(theta.flatten(order="F")),
+                    theta.flatten(order="F"),
+                    zeta.flatten(order="F"),
+                )
+            ).T
+            trans_temp = Transform(
+                Grid(nodes, sort=False, jitable=True),
+                basis,
+                derivs=jnp.array([[0, 0, 1]]),
+            )
+            return trans_temp.transform(Phi_mn, dz=1) + G / 2 / jnp.pi
+
+        super().__init__(
+            potential=_fourier_potential,
+            surface_grid=surface_grid,
+            params={"I": self.I, "G": self.G, "Phi_mn": self.Phi_mn},
+            potential_dtheta=_fourier_potential_d_theta,
+            potential_dzeta=_fourier_potential_d_zeta,
+            R_lmn=R_lmn,
+            Z_lmn=Z_lmn,
+            modes_R=modes_R,
+            modes_Z=modes_Z,
+            NFP=NFP,
+            sym=sym,
+            rho=rho,
+            name=name,
+            check_orientation=check_orientation,
+            surface=surface,
+        )
+
+    @property
+    def I(self):  # noqa: E743
+        """Net current linking the plasma and the coils toroidally."""
+        return self._I
+
+    @I.setter
+    def I(self, new):  # noqa: E743
+        assert np.isscalar(new), "I must be a scalar"
+        self._I = new
+        self._params["I"] = new
```

Comment:
a bit confused, is this its own attribute or is it part of the params dict?

Reply:
params but added it as an attribute as well to allow checking of it being a scalar. Though this is bypassed when it is passed in as a param I guess

Reply:
I moved FOurierCurrentPotentialField to be its own class instead of subclass with CurrentPotentialFIeld, so this is now just an attribute